packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -428,14 +428,18 @@ class HotRunner extends ResidentRunner {
     for (FlutterDevice device in flutterDevices) {
       // VM must have accepted the kernel binary, there will be no reload
       // report, so we let incremental compiler know that source code was accepted.
-      if (device.generator != null)
+      if (device.generator != null) {
         device.generator.accept();
+      }
     }
     // Check if the isolate is paused and resume it.
     final List<Future<void>> futures = <Future<void>>[];
+    // Only attempt to restart or reload the main isolate.
+    // The name of the ui isolate will be something like [file].dart$[entrypoint]-number
```

Comment:
This isn't necessarily true as the VM doesn't provide a guarantee for the format of isolate names (see [this reverted CL](https://dart-review.googlesource.com/c/sdk/+/90341/6/runtime/observatory/tests/service/set_name_rpc_test.dart) which is going to be relanded shortly).

Reply:
We'll need a way to match up an isolate to the entrypoint we've compiled. Is there a vmservice method which will tell us the URI of an isolate's entrypoint or similar?

Reply:
I don't believe so. However, there is couple unique identifiers in the `IsolateRef` object in the service protocol: https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#isolate.